### ISSUE TITLE
Android map engine release question

### ISSUE THREAD

**Issue by AriesJiang** [mapsme/omim#2142]
Hi, when switched activity from "MwmActivity"，method "MapFragment.nativeDestroyEngine()" spend too much time sometimes. So how do I solve this problem, or do you have any suggestions.Thx!

**Comment by trashkalmar** [comment]
@AriesJiang yep, this is true for debug version. If you experience the same issue on release build, please give us details about your device.

**Comment by AriesJiang** [comment]
It happens in some low performance machine for release version, such as, Moto g（XT1079）16GB.So how do I solve this?

**Comment by deathbaba** [comment]
@rokuz Can we do something about it? Why do we wait there? What does profiler shows?

**Comment by rokuz** [comment]
It can be more obvious on low performance machines and debug version. We must wait for destroying all OpenGL resources. You will get a crash otherwise.
